hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapOperation.java

```diff
@@ -98,12 +111,23 @@ public final void beforeRun() throws Exception {
         }
 
         canPublishWanEvent = canPublishWanEvent(mapContainer);
+        // TODO: add mapConfig#offload parameter here
+        hasMapStore = recordStore != null;
+        MapConfig mapConfig = mapContainer.getMapConfig();
+        mapStoreOffloadEnabled = getStartingStep() != null
+                && hasMapStore
+                && !mapConfig.getTieredStoreConfig().isEnabled();
```

Comment:
What does tiered store has to do with enabling / disabling offload?

Reply:
it has its own thread marking before and after runs and also it will not be permitted to run both map-store and tiered-store on the same map.

Reply:
>  it will not be permitted to run both map-store and tiered-store on the same map.
why not? it doesn't make sense.

Reply:
I was referring this discussion https://hazelcast.slack.com/archives/C01PD4KN9FB/p1635335337046200, but seems my interpretation of it can be not true since there seems no final decision yet. But let's address this separately in another PR.

Reply:
Further info: I had a look at current code base and tiered-store feature is coded as it expects no map-store interaction. So for now it is safe to assume there will be no map-store when tiered-store is enabled.